src/main/java/cz/student/ondrejmal/TaskManager.java

```diff
@@ -1,9 +1,138 @@
+
+        String filePath = "tasks.csv";
+        ArrayList<String[]> tasks = loadTasks(filePath);
+
+
+        Scanner scanner = new Scanner(System.in);
+
+        printOptions();
+
+        while (true) {
+            String command = scanner.nextLine().trim().toLowerCase();
+            System.out.println("Your command is: " + command.toUpperCase());
+
+            switch (command) {
+                case "add" -> addingTask(tasks, scanner);
+                case "remove" -> removingTask(tasks, scanner);
+                case "list" -> listingTasks(tasks);
+                case "exit" -> programExit(tasks, filePath, scanner);
+                default -> System.out.println("Invalid task selection. Please enter a valid task");
+            }
+        }
+    }
 
+    public static void printOptions() {
+        System.out.println("Please type name of possible operations to proceed.");
+        System.out.println("Select the task from the list below:");
+        System.out.println("1) Add");
+        System.out.println("2) Remove");
+        System.out.println("3) List");
+        System.out.println("4) Exit");
+        System.out.println("Command in not case sensitive.");
     }
 
-}
+    public static ArrayList<String[]> loadTasks(String filepath) {
+        ArrayList<String[]> fileTasks = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(filepath))) {
+            String line;
+            while ((line = reader.readLine()) != null)
+                fileTasks.add(line.split(","));
+        } catch (
+                IOException e) {
+            System.err.println("Problem with loading the file. File not found. Check if the file exists.");
+            e.printStackTrace(System.err);
+        }
+        return fileTasks;
+    }
+
+    public static void addingTask(ArrayList<String[]> tasks, Scanner scanner) {
+
+        System.out.println("Enter note:");
+        String note = scanner.nextLine();
+        while (note.isEmpty()) {
+            System.out.println("Please enter text to create a note.");
+            note = scanner.nextLine();
+        }
+
+        System.out.println("Enter date in YYYY-MM-DD format:");
+        String date = scanner.nextLine();
+        while (!date.matches("\\d{4}-\\d{2}-\\d{2}")) {
+            System.out.println("Format is not valid. Please enter the value again.");
+            date = scanner.nextLine();
+        }
+
+        System.out.println("Enter true/false value for statement of task completion:");
+        String statusInput = scanner.nextLine();
+
+        while (!statusInput.equalsIgnoreCase("true") && !statusInput.equalsIgnoreCase("false")) {
+            System.out.println("Invalid input. Please enter true or false:");
+            statusInput = scanner.nextLine();
+        }
+        boolean status = Boolean.parseBoolean(statusInput);
+
+        tasks.add(new String[]{note, date, String.valueOf(status)});
```

Comment:
CLEAN-CODE: kdyz se na ten kod divam tak prevadis status na boolean a pak zpatky  na string, tohle bych zrusil.

```suggestion
        tasks.add(new String[]{note, date,statusInput});
```

Reply:
opraveno